pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -271,42 +260,28 @@ type BuildList struct {
 }
 
 // GetCondition returns the Condition matching the given type.
-func (bs *BuildStatus) GetCondition(t BuildConditionType) *BuildCondition {
-	for _, cond := range bs.Conditions {
-		if cond.Type == t {
-			return &cond
-		}
-	}
-	return nil
+func (bs *BuildStatus) GetCondition(t duckv1alpha1.ConditionType) *duckv1alpha1.Condition {
+	return buildCondSet.Manage(bs).GetCondition(t)
 }
 
 // SetCondition sets the condition, unsetting previous conditions with the same
 // type as necessary.
-func (b *BuildStatus) SetCondition(newCond *BuildCondition) {
-	if newCond == nil {
-		return
+func (bs *BuildStatus) SetCondition(newCond *duckv1alpha1.Condition) {
```

Comment:
you should be able to remove this entirely and replace it with `Mark*` methods. Looks like there is only Succeeded and Failed. I would implement:

`MarkBuildSucceded`
`MarkBuildFailed`

and then remove SetCondition entirely.

Reply:
Ack.  Build hasn't yet adopted this model, so I think that should be a follow up.